examples/applications/author-disambiguation/distance.py

```diff
@@ -0,0 +1,233 @@
+                ("shaper", Shaper(newshape=(-1,))),
+                ("tf-idf", TfidfVectorizer(dtype=np.float32,
+                                           decode_error="replace")),
+            ]), groupby=group_by_signature)),
+            ("combiner", CosineSimilarity())
+        ])),
+        ("collaborations_similarity", Pipeline([
+            ("pairs", PairTransformer(element_transformer=Pipeline([
+                ("collaborations", FuncTransformer(func=get_collaborations)),
+                ("shaper", Shaper(newshape=(-1,))),
+                ("tf-idf", TfidfVectorizer(dtype=np.float32,
+                                           decode_error="replace")),
+            ]), groupby=group_by_signature)),
+            ("combiner", CosineSimilarity())
+        ])),
+        ("references_similarity", Pipeline([
+            ("pairs", PairTransformer(element_transformer=Pipeline([
+                ("references", FuncTransformer(func=get_references)),
+                ("shaper", Shaper(newshape=(-1,))),
+                ("tf-idf", TfidfVectorizer(dtype=np.float32,
+                                           decode_error="replace")),
+            ]), groupby=group_by_signature)),
+            ("combiner", CosineSimilarity())
+        ])),
+        ("year_diff", Pipeline([
+            ("pairs", FuncTransformer(func=get_year, dtype=np.int)),
+            ("combiner", AbsoluteDifference())
+        ]))])
+
+    # Train a classifier on these vectors
+    classifier = GradientBoostingClassifier(n_estimators=500,
+                                            max_depth=9,
+                                            max_features=10,
+                                            learning_rate=0.125,
+                                            verbose=3)
+
+    # Return the whole pipeline
+    estimator = Pipeline([("transformer", transformer),
+                          ("classifier", classifier)]).fit(X, y)
+
+    return estimator
+
+
+def learn_model(distance_pairs, input_signatures, input_records,
+                distance_model):
+    """Learn the distance model using the files from the disc.
+
+    Parameters
+    ----------
+    :param distance_pairs: string
+        Path to the file with distance pairs. The content should be a json
+        array of tuples (`signature_id1`, `signature_id2`, `target`),
+        where `target = 0` if both signatures belong to the same author,
+        and `target = 1` otherwise.
+
+        [(0, 1, 0), (2, 3, 0), (4, 5, 1), ...]
+
+    :param input_signatures: string
+        Path to the file with signatures. The content should be a json array
+        of dictionaries holding metadata about signatures.
+
+        [{"signature_id": 0,
+          "author_name": "Doe, John",
+          "publication_id": 10, ...}, { ... }, ...]
+
+    :param input_records: string
+        Path to the file with records. The content should be a json array of
+        dictionaries holding metadata about records
+
```

Comment:
need to complete the Parameters documentation for 
: param distance_model: string
we need to say that it have to be something.cPickle